File: python/udp_server.py
import time

from support_loader import espp

start = time.time()
def on_receive_data(data, sender_info):
    global start
    port = 5555
    elapsed = time.time() - start
    print(f"[{elapsed:.3f}] Received: '{data}'")
    str_data = ''.join(chr(x) for x in data)
    print(f"       as string: '{str_data}'")
    print(f"            from: {sender_info.address}:{sender_info.port}")
    ret_data = list("Data returned to sender goes here...")
    print(f"      responding: {len(ret_data)} bytes to {sender_info.address}:{sender_info.port}")
    # turn the list of characters into a list of integers
    ret_data = [ord(x) for x in ret_data]
    return ret_data

udp_client = espp.UdpSocket(espp.UdpSocket.Config(espp.Logger.Verbosity.debug))
port = 5555
buffer_size = 1024
receive_config = espp.UdpSocket.ReceiveConfig(
    port,
    buffer_size,
    False,
    '',
    on_receive_data
)
udp_client.start_receiving(espp.Task.Config("udp_task", None), receive_config)

time.sleep(10)

exit()

File: pc/tests/udp_server.cpp
#include "espp.hpp"

using namespace std::chrono_literals;

static auto start = std::chrono::high_resolution_clock::now();

int main() {
  static auto server = espp::UdpSocket({.log_level = espp::Logger::Verbosity::DEBUG});
  auto server_task_config = espp::Task::Config{
      .name = "UdpServer",
      .callback = nullptr,
      .stack_size_bytes = 6 * 1024,
  };

  auto server_config = espp::UdpSocket::ReceiveConfig{
      .port = 5555, .buffer_size = 1024, .on_receive_callback = [
      ](auto &data, auto &source) -> auto{auto now = std::chrono::high_resolution_clock::now();
  auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(now - start).count();
  fmt::print("Received {} bytes from {}:{}\n", data.size(), source.address, source.port);
  auto message =
      fmt::format("Received {} bytes from {}:{}\n", data.size(), source.address, source.port);
  // convert the message into a vector of bytes
  std::vector<uint8_t> message_bytes(message.begin(), message.end());
  // send the message back to the client
  return message_bytes;
}
}
;
server.start_receiving(server_task_config, server_config);

std::this_thread::sleep_for(10s);

return 0;
}
